### ISSUE TITLE
Function.name doesnt work in IE10+

### ISSUE THREAD

**Issue by nickcolley** [pouchdb/pouchdb#4216]
https://github.com/pouchdb/pouchdb/blob/70761f2c6003e91f8a886a1d5968c34cb27dc047/lib/adapters/leveldb/index.js#L150

leveldown.name is set only in browsers that support .name on a function

http://stackoverflow.com/questions/6903762/function-name-not-supported-in-ie

**Comment by nickcolley** [comment]
I wacked this polyfill in:
```
if (!(function f() {}).name) {
    Object.defineProperty(Function.prototype, 'name', {
        get: function() {
            var name = this.toString().match(/^\s*function\s*(\S*)\s*\(/)[1];
            // For better performance only parse once, and then cache the
            // result through a new accessor for repeated access.
            Object.defineProperty(this, 'name', { value: name });
            return name;
        }
    });
}
```

Memdown passes in IE! :D

Might be a nicer way to fix without polyfilling.

**Comment by nickcolley** [comment]
This is again ES6... https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name

**Comment by nickcolley** [comment]
This is required for pouchdb.memory to work in IE by the way, didn't make that clear.

**Comment by nolanlawson** [comment]
So the thing is that `Function.name` is really a hack, and we are exploiting the fact that most of the *down adapters have names like `MemDOWN` and `LocalStorageDOWN` that make them easy to identify. But in fact there's nothing stopping them from exporting anonymous functions if they feel like it.

I'd suggest that instead of providing a `Function.name` shim, we just rewrite the part of the code that checks for it.

**Comment by nolanlawson** [comment]
fixed by 2b656465bd578de1180c5b50d73664935a1f37fe

**Comment by nolanlawson** [comment]
finally actually for-realseys fixed by 5c0c421081472a785b5bee58860420c57aae8885
